ShopAppTests/App/Account/ViewModels/AddressFormViewModelSpec.swift

```diff
@@ -0,0 +1,113 @@
+        
+        describe("when view model initialized") {
+            it("should have variables with correct initial values") {
+                expect(viewModel.address).to(beNil())
+            }
+        }
+        
+        describe("when countries loaded") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+            }
+            
+            it("should return correct contry names") {
+                viewModel.namesOfCountries
+                    .subscribe(onNext: { names in
+                        expect(names) == ["Country1", "Country2"]
+                    })
+                    .disposed(by: disposeBag)
+                
+                viewModel.getCountries()
+            }
+        }
+        
+        describe("when country updated") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+            }
+            
+            context("if states not found") {
+                it("should return empty array") {
+                    countriesUseCaseMock.returnStates = false
+                    viewModel.getCountries()
+                    
+                    viewModel.namesOfStates
+                        .subscribe(onNext: { names in
+                            expect(names.isEmpty) == true
+                        })
+                        .disposed(by: disposeBag)
+                    
+                    viewModel.updateStates(with: "Country1")
+                }
+            }
+            
+            context("if states found") {
+                it("should return array of states") {
+                    countriesUseCaseMock.returnStates = true
+                    viewModel.getCountries()
+                    
+                    viewModel.namesOfStates
+                        .subscribe(onNext: { names in
+                            expect(names.isEmpty) == false
+                        })
+                        .disposed(by: disposeBag)
+                    
+                    viewModel.updateStates(with: "Country1")
+                }
+            }
+        }
+        
+        describe("when submit button did press") {
+            var disposeBag: DisposeBag!
+            var address: Address!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                address = Address()
+                address.country = "Country"
+                address.firstName = "First name"
+                address.lastName = "Last name"
+            }
+            
+            it("should return address with correct fields") {
+                viewModel.address = address
+                viewModel.updateFields()
+                
+                viewModel.filledAddress
+                    .subscribe(onNext: { filledAddress in
+                        expect(filledAddress.country) == address.country
```

Comment:
I think we need to cover all changes so test all variables that can be changed by `updateFields` method call.